### PR TITLE
[RFC] plug document interface and clean up missing errors

### PR DESCRIPTION
**Pull request by cordoval**

### CODE REVIEW

**Comment by cordoval** [comment]
you may merge early since the tests i bet are not passing anyways, because of the errors of prices and such
but if not please comment the PR so it can be guided :+1:  :baby:

**Comment by wjzijderveld** [comment]
Why did you remove the InvoiceItem? Doesn't seem like something that we can just lose.

**Comment by cordoval** [comment]
@wjzijderveld because is a duplicate minus two simple methods of the other Order/Item I merged them

very confusing and error prone to have classes of the same name don't you think?

**Comment by wjzijderveld** [comment]
I agree that it can be confusing. Just wondering how you would proceed when you wouldn't have an order but would only create invoices. 
I think we should have the separate objects, but maybe we need to move a lot of the duplicate stuff to a BaseSalesItem or something like that?

**Comment by inspiran** [comment]
I do agree with Willem that both order and invoice item interface might  share a base interface (and the same for the class) but that should be about it;  Both entities serve a different purpose in a whole.  Often it might be the case that it is a 1:1 copy (for simple ecommerce cart->order->invoice processes) but in other cases they will behave differently.   

For instance if you allow a customer to place multiple orders through a store and only invoice once a month, the relationship between an order item and invoice item will be not 1:1  and typically no order related data is needed on the invoice item;  Hence they should be generalized except some methods (eg. the item number, item description, item prices)

**Comment by wjzijderveld** [comment]
I only meant this for this entity. The persistence layer should of course not share the information.
But for the pure PHP model, I see no harm in moving duplicate properties to a base class? Or am I missing something else?

**Comment by iampersistent** [comment]
Actually @cordoval the failed test is a legitimate failure https://travis-ci.org/vespolina/VespolinaCore/jobs/9973564 he namespace needs to be update. Core has been passing https://travis-ci.org/vespolina/VespolinaCore/jobs/9928693 although the tests need to be targeting 5.5.

I agree with a shared interface for Order\Item and Invoice\Items and I'm also ok with a shared interface between Order and Invoice. I have to say I"m not fond of the name DocumentInterface. To me, a Document is like a PDF, ODF or DOCX. I don't necessarily have a better name at this point. Maybe ItemableInterface or something along those lines.

**Comment by iampersistent** [comment]
I'm with @wjzijderveld, I see no harm in a base class.

**Comment by iampersistent** [comment]
@cordoval I added a couple of notes on the DocumentInterface. 
It might be good to also have an abstract Itemable class that has the common code for adding Items that is inherited by Vespolina\Entity\Order\Order and Vespolina\Entity\Invoice\Invoice Do you mind handling that?

Additionally, would you be ok with adding a Vespolina\Entity\ItemInterface that will be inherited by Vespolina\Entity\Order\ItemInterface and Vespolina\Entity\Invoice\ItemInterface. Along with that, create an  Vespolina\Entity\Item class with the common code between Vespolina\Entity\Order\Item and Vespolina\Entity\Invoice\Item.

**Comment by cordoval** [comment]
I will adapt as discussed just on the morrow :baby:

**Comment by cordoval** [comment]
@iampersistent ready to merge

File: lib/Vespolina/Entity/ItemableInterface.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ *
+ * @author Luis Cordova <cordoval@gmail.com>
+ */
+interface ItemableInterface
+{
+    function addItem(ItemInterface $item);
+
+    function removeItem(ItemInterface $item);
+
+    function clearItems();
```
Comment (wjzijderveld) [inline lib/Vespolina/Entity/ItemableInterface.php:23]: Please add some docblocks above this methods

**Pull request merged**